cmd/kubeadm/app/cmd/phases/bootstraptoken.go

```diff
@@ -0,0 +1,139 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package phases
+
+import (
+	"fmt"
+
+	"github.com/spf13/cobra"
+
+	clientset "k8s.io/client-go/kubernetes"
+	"k8s.io/kubernetes/cmd/kubeadm/app/phases/bootstraptoken/clusterinfo"
+	"k8s.io/kubernetes/cmd/kubeadm/app/phases/bootstraptoken/node"
+	kubeadmutil "k8s.io/kubernetes/cmd/kubeadm/app/util"
+	kubeconfigutil "k8s.io/kubernetes/cmd/kubeadm/app/util/kubeconfig"
+	versionutil "k8s.io/kubernetes/pkg/util/version"
+)
+
+// NewCmdBootstrapToken returns the Cobra command for running the mark-master phase
+func NewCmdBootstrapToken() *cobra.Command {
+	var kubeConfigFile string
+	cmd := &cobra.Command{
+		Use:     "bootstrap-token",
+		Short:   "Manage kubeadm-specific Bootstrap Token functions.",
+		Aliases: []string{"bootstraptoken"},
+		RunE:    subCmdRunE("bootstrap-token"),
+	}
+
+	cmd.PersistentFlags().StringVar(&kubeConfigFile, "kubeconfig", "/etc/kubernetes/admin.conf", "The KubeConfig file to use for talking to the cluster")
+
+	// Add subcommands
+	cmd.AddCommand(NewSubCmdClusterInfo(&kubeConfigFile))
+	cmd.AddCommand(NewSubCmdNodeBootstrapToken(&kubeConfigFile))
+
+	return cmd
+}
+
+// NewSubCmdClusterInfo returns the Cobra command for running the cluster-info sub-phase
+func NewSubCmdClusterInfo(kubeConfigFile *string) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:     "cluster-info <clusterinfo-file>",
+		Short:   "Uploads and exposes the cluster-info ConfigMap publicly from the given cluster-info file",
+		Aliases: []string{"clusterinfo"},
+		Run: func(cmd *cobra.Command, args []string) {
+			err := validateExactArgNumber(args, []string{"clusterinfo-file"})
+			kubeadmutil.CheckErr(err)
+
+			client, err := kubeconfigutil.ClientSetFromFile(*kubeConfigFile)
+			kubeadmutil.CheckErr(err)
+
+			// Here it's safe to get args[0], since we've validated that the argument exists above in validateExactArgNumber
+			clusterInfoFile := args[0]
```

Comment:
That is not required as kubeadm just strips out and unnamed or currently used cluster from that file: https://github.com/kubernetes/kubernetes/blob/master/cmd/kubeadm/app/util/kubeconfig/kubeconfig.go#L103

So it will never send creds. Not sure if we have to restrict having no creds here...